datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/dataset/ActorUsersResolver.java

```diff
@@ -0,0 +1,71 @@
+package com.linkedin.datahub.graphql.resolvers.dataset;
+
+import static com.linkedin.datahub.graphql.resolvers.ResolverUtils.bindArgument;
+import static com.linkedin.metadata.Constants.CORP_USER_ENTITY_NAME;
+
+import com.linkedin.common.urn.Urn;
+import com.linkedin.common.urn.UrnUtils;
+import com.linkedin.datahub.graphql.QueryContext;
+import com.linkedin.datahub.graphql.generated.CorpUser;
+import com.linkedin.datahub.graphql.generated.RoleUser;
+import com.linkedin.datahub.graphql.generated.UserFilter;
+import com.linkedin.datahub.graphql.types.corpuser.mappers.CorpUserMapper;
+import com.linkedin.entity.EntityResponse;
+import com.linkedin.entity.client.EntityClient;
+import com.linkedin.metadata.query.ListResult;
+import com.linkedin.r2.RemoteInvocationException;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.net.URISyntaxException;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequiredArgsConstructor
+public class ActorUsersResolver implements DataFetcher<CompletableFuture<ArrayList<RoleUser>>> {
+
+  private final EntityClient _client;
+
+  @Override
+  public CompletableFuture<ArrayList<RoleUser>> get(final DataFetchingEnvironment environment)
+      throws Exception {
+    final UserFilter userFilter = bindArgument(environment.getArgument("input"), UserFilter.class);
+
+    final QueryContext context = environment.getContext();
+
+    return CompletableFuture.supplyAsync(
+        () -> {
+          try {
+            if (userFilter != null) {
+              Urn userUrn = UrnUtils.getUrn(userFilter.getUserUrn());
+              EntityResponse userEntityResponse =
+                  _client.getV2(CORP_USER_ENTITY_NAME, userUrn, null, context.getAuthentication());
+              if (userEntityResponse == null) {
+                return null;
+              }
+              CorpUser corpUser = CorpUserMapper.map(userEntityResponse);
+              RoleUser roleUser = new RoleUser();
+              roleUser.setUser(corpUser);
+              ArrayList<RoleUser> roleUsers = new ArrayList<>();
+              roleUsers.add(roleUser);
+              return roleUsers;
```

Comment:
so the way i'm reading this logic is that if you have a userFilter with a user urn in it, simply fetch the user and populate it in the list of users. Wouldn't this be incorrect, though, if the user is not in this role? with this filter the user you provide would always be in the list of every role you give.

I think we need a different approach to improving this performance here.